### PR TITLE
Exponential penalty implementing

### PR DESCRIPTION
**Pull request by sandrodevdariani**
### **PR REVIEW:** 

1. create new file for exponential penalty **neurons/validators/penalty/exponential_penalty.py** and add class **ExponentialTimePenaltyModel**

**functionality** - The **ExponentialTimePenaltyModel** calculates penalties for response times using exponential logic. For responses faster than a maximum execution time, it applies smaller penalties, rewarding efficiency. The class integrates dynamic thresholds using get_max_execution_time, default model is NOVA

2.  Update **ScraperValidator** initialize **ExponentialTimePenaltyModel**  in self.penalty_functions
3. Add simple checker for events in config.py as we have exponential and streaming penalties now.
4.  Update **validator.py** add that line of code  
```
while True:
   await asyncio.sleep(10) 
```
without that Validator was keep shutting down by itself.

### CODE REVIEW

File: datura/__init__.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 
 # version must stay on line 22
-__version__ = "0.0.162"
+__version__ = "0.0.163"
```
Comment (Chkhikvadze) [inline datura/__init__.py:22]: Don't update the version yet; we will update it when we move to Finney.

File: neurons/validators/penalty/exponential_penalty.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        for index, response in enumerate(responses):
+            process_time = getattr(response.dendrite, "process_time", None)
+
+            if process_time is None:
+                # Apply maximum penalty if process_time is missing
+                bt.logging.warning(
+                    f"Process time is missing for response {index}, applying maximum penalty."
+                )
+                penalties[index] = MAX_PENALTY
```
Comment (Chkhikvadze) [inline neurons/validators/penalty/exponential_penalty.py:41]: Why do you set penalties[index] = MAX_PENALTY if the process time is none?

File: neurons/validators/validator.py [hunk 3]
```diff
@@ -496,6 +496,12 @@ async def run_organic_with_interval(interval):
 
             three_hours_in_seconds = 10800
             self.loop.create_task(run_organic_with_interval(three_hours_in_seconds))
+
+            # without that validator exit itself. 
+            while True:
```
Comment (Chkhikvadze) [inline neurons/validators/validator.py:501]: Why did you add it??

**Review by Chkhikvadze — CHANGES_REQUESTED** [review]

File: datura/__init__.py (hunk 1 above)
Comment (sandrodevdariani) [reply datura/__init__.py:22]: its not my commit : https://github.com/Datura-ai/smart-scrape/pull/195/commits/351b8f689d29182eb980bd47d51c6ab50af92c7c
